File: clients/javascript/src/apis/rateLimiter.ts
// this file is @generated

import {
    type RateLimiterCheckIn,
    RateLimiterCheckInSerializer,
} from '../models/rateLimiterCheckIn';
import {
    type RateLimiterCheckOut,
    RateLimiterCheckOutSerializer,
} from '../models/rateLimiterCheckOut';
import {
    type RateLimiterGetRemainingIn,
    RateLimiterGetRemainingInSerializer,
} from '../models/rateLimiterGetRemainingIn';
import {
    type RateLimiterGetRemainingOut,
    RateLimiterGetRemainingOutSerializer,
} from '../models/rateLimiterGetRemainingOut';
import { RateLimiterNamespace } from './rateLimiterNamespace';
import { HttpMethod, CoyoteRequest, type CoyoteRequestContext } from "../request";

export class RateLimiter {
    public constructor(private readonly requestCtx: CoyoteRequestContext) {}

    public get namespace() {
        return new RateLimiterNamespace(this.requestCtx);
    }

    /** Rate Limiter Check and Consume */
        public limit(
            rateLimiterCheckIn: RateLimiterCheckIn,
            ): Promise<RateLimiterCheckOut> {
            const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/rate-limit/limit");

            request.setBody(
                    RateLimiterCheckInSerializer._toJsonObject(
                        rateLimiterCheckIn,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    RateLimiterCheckOutSerializer._fromJsonObject,
                );
            }

        

    /** Rate Limiter Get Remaining */
        public getRemaining(
            rateLimiterGetRemainingIn: RateLimiterGetRemainingIn,
            ): Promise<RateLimiterGetRemainingOut> {
            const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/rate-limit/get-remaining");

            request.setBody(
                    RateLimiterGetRemainingInSerializer._toJsonObject(
                        rateLimiterGetRemainingIn,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    RateLimiterGetRemainingOutSerializer._fromJsonObject,
                );
            }

        

    }

File: clients/java/src/main/java/com/svix/diom/apis/RateLimiter.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.ApiException;
import com.svix.diom.HttpClient;
import com.svix.diom.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.diom.models.RateLimiterCheckIn;
import com.svix.diom.models.RateLimiterCheckOut;
import com.svix.diom.models.RateLimiterGetRemainingIn;
import com.svix.diom.models.RateLimiterGetRemainingOut;

public class RateLimiter {
    private final HttpClient client;

    public RateLimiter(HttpClient client) {
        this.client = client;
    }

    /** Rate Limiter Check and Consume */
    public RateLimiterCheckOut limit(
        final RateLimiterCheckIn rateLimiterCheckIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/rate-limiter/limit");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            rateLimiterCheckIn,
            RateLimiterCheckOut.class
            );
    }

    /** Rate Limiter Get Remaining */
    public RateLimiterGetRemainingOut getRemaining(
        final RateLimiterGetRemainingIn rateLimiterGetRemainingIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/rate-limiter/get-remaining");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            rateLimiterGetRemainingIn,
            RateLimiterGetRemainingOut.class
            );
    }
}